src/vmemcache.c

```diff
@@ -406,26 +436,32 @@ vmemcache_get(VMEMcache *cache, const void *key, size_t ksize, void *vbuf,
 	struct cache_entry *entry;
 	size_t read = 0;
 
+	util_fetch_and_add64(&cache->get_count, 1);
+
 	int ret = vmcache_index_get(cache->index, key, ksize, &entry);
 	if (ret < 0)
 		return -1;
 
 	if (entry == NULL) { /* cache miss */
 		util_fetch_and_add64(&cache->miss_count, 1);
 
-		if (cache->on_miss == NULL ||
-		    (*cache->on_miss)(cache, key, ksize, cache->arg_miss))
-			return 0;
+		if (cache->on_miss) {
+			get_req.key = key;
+			get_req.ksize = ksize;
+			get_req.vbuf = vbuf;
+			get_req.vbufsize = vbufsize;
+			get_req.offset = offset;
+			get_req.vsize = vsize;
 
-		ret = vmcache_index_get(cache->index, key, ksize, &entry);
-		if (ret < 0)
-			return -1;
+			(*cache->on_miss)(cache, key, ksize, cache->arg_miss);
 
-		if (entry == NULL)
-			return 0;
-	}
+			if (!get_req.key)
+				return (ssize_t)get_req.vbufsize;
```

Comment:
Can be distinguish between copying 0 bytes and failing to find the element? I think this should return -1.